qabelbox/src/main/java/de/qabel/qabelbox/activities/CreateIdentityActivity.kt

```diff
@@ -0,0 +1,257 @@
+
+    override val wizardEntityLabel: String get() = getString(R.string.identity)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        QabelBoxApplication.getApplicationComponent(applicationContext).inject(this)
+        super.onCreate(savedInstanceState)
+        dropUrlGenerator = DropUrlGenerator(getString(R.string.dropServer))
+        identityUseCase.getIdentities().subscribe({
+            existingIdentities = it
+            canExit = it.identities.size > 0
+        }, {
+            existingIdentities = Identities()
+        })
+    }
+
+    private fun tryReadPhoneNumber() {
+        val permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.READ_PHONE_STATE)
+        if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.READ_PHONE_STATE)) {
+                alert(R.string.dialog_headline_info, R.string.phone_number_request_info, {
+                    positiveButton(R.string.yes) { tryReadPhoneNumber() }
+                    negativeButton(R.string.no) { READ_PHONE_STATE_DENIED = true }
+                })
+            } else {
+                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.READ_PHONE_STATE), REQUEST_READ_PHONE_STATE)
+            }
+        } else {
+            val phoneManager = ctx.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+            var phone = phoneManager.line1Number
+            if (phone != null && !phone.isEmpty()) {
+                try {
+                    phone = formatPhoneNumber(phone)
+                    (fragments[3] as CreateIdentityEditTextFragment).setValue(phone)
+                    info("Phone number detected $phoneNumber")
+                } catch (ex: NumberFormatException) {
+                    error("Error parsing received system phone number $phone", ex)
+                }
+            }
+        }
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == REQUEST_READ_PHONE_STATE) {
+            if (permissions.contains(Manifest.permission.READ_PHONE_STATE)) {
+                if (grantResults.contains(PackageManager.PERMISSION_GRANTED)) {
+                    tryReadPhoneNumber()
+                } else if (grantResults.contains(PackageManager.PERMISSION_DENIED)) {
+                    READ_PHONE_STATE_DENIED = true
+                }
+            }
+        }
+    }
+
+    override fun handleNextClick() {
+        super.handleNextClick()
+        if (step > 0 && tryCount != 3 && prefix == null) {
+            loadPrefixInBackground()
+        }
+        if (step == 2 && !READ_PHONE_STATE_DENIED) {
+            tryReadPhoneNumber()
+        }
+    }
+
+    override val fragments: Array<BaseIdentityFragment> =
```

Comment:
changed but not solved. You have more occurrences here https://github.com/Qabel/qabel-android/pull/645/files#diff-7aa47eb5b151ebe1cadcdaca4e9f8424R82
and you are still accessing fragments by index. It may be code that existed before but since you touched it and now have a by far better option, don't forget about the boy scout rule.